apps/address-book/app/components/App/App.js

```diff
@@ -1,106 +1,79 @@
-import { observe, SidePanel, Main, AppBar, AppView } from '@aragon/ui'
-import PropTypes from 'prop-types'
-import React from 'react'
-import { hot } from 'react-hot-loader'
+import React, { useState } from 'react'
 import styled from 'styled-components'
-import { map } from 'rxjs/operators'
-import Entities from './Entities'
-import NewEntity from '../Panel/NewEntity'
-import { networkContextType, AppTitle, AppTitleButton } from '../../../../../shared/ui'
-
-class App extends React.Component {
-  static propTypes = {
-    app: PropTypes.object.isRequired,
-    // TODO: Shape this
-    entities: PropTypes.arrayOf(PropTypes.object),
-  }
 
-  static defaultProps = {
-    network: {},
-  }
+import { useAragonApi } from '@aragon/api-react'
+import { AppBar, AppView, Main, SidePanel } from '@aragon/ui'
 
-  static childContextTypes = {
-    network: networkContextType,
-  }
+import { AppTitle, AppTitleButton } from '../../../../../shared/ui'
+import { IdentityProvider } from '../../../../../shared/identity'
+import Entities from './Entities'
+import NewEntity from '../Panel/NewEntity'
 
-  state = {
-    panelVisible: false,
+const ASSETS_URL = './aragon-ui'
+
+const App = () => {
+  const [ panelVisible,setPanelVisible ] = useState(false)
+  const { api, appState = {}, displayMenuButton = false } = useAragonApi()
+  
+  const { entries = [] } = appState
+  
+  const createEntity = ({ address, name, type }) => {
+    api.addEntry(address, name, type)
+    closePanel()
   }
 
-  getChildContext() {
-    const { network } = this.props
-    return {
-      network: {
-        type: network.type,
-      },
-    }
+  const removeEntity = (address) => {
+    api.removeEntry(address)
```

Comment:
No need for this intermediate function. We can pass `onRemoveEntity={api.removeEntry}` instead.